spikeextractors/extractors/neoextractors/neobaseextractor.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+import neo
+
+from ..recordingextractor import RecordingExtractor
+from ..sortingextractor import SortingExtractor
+
+
+class NeoBaseRecordingExtractor(RecordingExtractor):
+    NeoRawIOClass = None
+    def __init__(self, **kargs):
+        self.neo_reader = self.NeoRawIOClass(**kkargs)
+        self.neo_reader.parse_header()
+        
+        # TODO propose a meachanisim to select the appropriate segment
+        # in case there are several
+        num_block = self.neo_reader.block_count()
+        assert num_block == 1, 'This file is multi block spikeextractors support only one segment'
+        num_seg = self.neo_reader.segment_count()
+        assert num_seg == 1, 'This file is multi segment spikeextractors support only one segment'
+        
+        # select first block first segment
+        self.block_index = 0
+        self.seg_index = 0
+        
+        # TODO propose a meachanisim to select the appropriate channel groups
+        # in neo one channel group have the same dtype/sampling_rate/group_id
+        num_chan_group = self.neo_reader.get_group_channel_indexes()
```

Comment:
Is this different than having channel groups as a spikeextractors property? can you have channel groups that belong to different recordings with different sampling_rates?

Reply:
No channel groups inside spiekinterface is a differents level.
In SI it is for grouping channels (like tetrodes).
At the rawio level, it is for sampling_rate and durations.
It is valid to have a channel shorter than another one (some recordings system able this!!).

So this ensure that a "neo group" is same sampling_rate/dtype/length wich is needed for a recording extractor.

Reply:
I see. Good to know.